tools/integration_tests/list_large_dir/list_dir_with_twelve_thousand_files_test.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package list_large_dir_test
+
+import (
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/setup"
+)
+
+func checkIfObjNameIsCorrect(objName string, prefix string, t *testing.T) (objNumber int) {
+	// Extracting file number.
+	objNumberStr := strings.ReplaceAll(objName, prefix, "")
+	objNumber, err := strconv.Atoi(objNumberStr)
+	if err != nil {
+		t.Errorf("Error in extracting file number.")
+	}
+	return
+}
+
+func throwErrorForIncorrectFileNumber(fileNumber int, t *testing.T) {
+	if fileNumber < 1 || fileNumber > 12000 {
```

Comment:
you can pass the max number (100 & 12000 in these cases) as argument to checkIfObjNameIsCorrect and add this check there. That way you don't need to have throwErrorForIncorrectFileNumber & throwErrorForIncorrectDirNumber